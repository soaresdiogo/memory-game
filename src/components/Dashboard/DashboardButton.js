import React from 'react';
import Button from '@material-ui/core/Button';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const DashboardButton = ({ children, to }) => (
  <Button
    data-testid="dashboardButton"
    component={Link}
    to={to}
    variant="outlined"
    color="primary"
    size="large"
    fullWidth
  >
    {children}
  </Button>
);

DashboardButton.defaultProp = {
  to: PropTypes.string,
};

DashboardButton.propTypes = {
  children: PropTypes.string.isRequired,
};

export default DashboardButton;
