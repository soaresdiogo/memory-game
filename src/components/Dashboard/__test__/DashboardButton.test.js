import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import DashboardButton from '../DashboardButton';

import '@testing-library/jest-dom/extend-expect';

afterEach(cleanup);

const Button = (
  <Router>
    <Route>
      <DashboardButton to="/game">Game Start</DashboardButton>
    </Route>
  </Router>
);

describe('Dashboard button component', () => {
  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(Button, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('should have the text Game Start', () => {
    const { getByTestId } = render(Button);
    expect(getByTestId('dashboardButton')).toHaveTextContent('Game Start');
  });

  test('matches snapshot', () => {
    const tree = renderer.create(Button).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
