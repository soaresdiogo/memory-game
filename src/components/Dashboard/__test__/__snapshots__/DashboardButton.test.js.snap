// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard button component matches snapshot 1`] = `
<a
  aria-disabled={false}
  className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-outlinedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
  data-testid="dashboardButton"
  href="/game"
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex={0}
>
  <span
    className="MuiButton-label"
  >
    Game Start
  </span>
</a>
`;
